File: Day_4/morse.py
"""
   Day 4: Morse Code Translator

   This is a program to translate english to
   morse code. It explores the world of functions,
   switch statements, and exceptions.
"""

# This allows the user to choose an option
# for the translator
def menu():
   choice = int(input("Morse Code Translator"+\
                       "\n_____________________"+\
                       "\n\t1. Translate character to Morse Code"+\
                       "\n\t2. Translate string to Morse Code"+\
                       "\n\t3.Exit"+\
                       "\nChoice: "))

   # No error checking, just end the program if
   # the user is being annoying
   if choice != 1 and choice != 2 and choice != 3:
      print("Invalid input: Quitting...")
      quit()

   return choice

# Translate to morse code
def english2morse(eng):
   res=''
   for c in eng.lower():
      if c=='a':   res+="·‒"
      elif c=='b': res+="‒···"
      elif c=='c': res+="‒·‒·"
      elif c=='d': res+="‒··"
      elif c=='e': res+="·"
      elif c=='f': res+="··‒·"
      elif c=='g': res+="‒‒·"
      elif c=='h': res+="····"
      elif c=='i': res+="··"
      elif c=='j': res+="·‒‒‒"
      elif c=='k': res+="‒·‒"
      elif c=='l': res+="·‒··"
      elif c=='m': res+="‒‒"
      elif c=='n': res+="‒·"
      elif c=='o': res+="‒‒‒"
      elif c=='p': res+="·‒‒·"
      elif c=='q': res+="‒‒·‒"
      elif c=='r': res+="·‒·"
      elif c=='s': res+="···"
      elif c=='t': res+="‒"
      elif c=='u': res+="··‒"
      elif c=='v': res+="···‒"
      elif c=='w': res+="·‒‒"
      elif c=='x': res+="‒··‒"
      elif c=='y': res+="‒·‒‒"
      elif c=='z': res+="‒‒··"
      else: res+=' '
   return res


stop=False
while not stop:
   choice=menu()
   # Input character, print morse
   if choice == 1:
      try:
         c=input("\nEnter a character: ")
         if not c.isalpha() or len(c) != 1:
            raise Exception()
      except Exception:
         print("Invalid input: Quitting...")
         quit()
      print("\nTranslation: " + english2morse(c))
   # Input string, print morse
   elif choice == 2:
      try:
         s=input("\nEnter a string: ")
         if not s.replace(' ','').isalpha():
            raise Exception()
      except Exception:
         print("Invalid input: Quitting...")
         quit()
      print("\nTranslation: " + english2morse(s))
   elif choice == 3:
      stop=True

File: Day_4/morse.java
/*
   Day 4: Morse Code Translator

   This is a program to translate english to
   morse code. It explores the world of functions,
   switch statements, and exceptions.
*/
import java.util.Scanner;
import java.lang.*;

public class morse
{
   public static void main(String[] args)
   {
      Scanner scanner = new Scanner(System.in);
      Boolean stop = false;
      while(!stop)
      {
         switch(menu())
         {
            // Input character, print morse
            case 1: Character c;
                    System.out.print("\nEnter a character: ");
                    try{
                       c=scanner.next().charAt(0);
                    }
                    catch(Exception e){
                       System.out.println("Invalid input: Quitting...");
                       return;
                    }
                    System.out.println("\nTranslation: "+english2morse(c));
                    break;
            // Input string, print morse
            case 2: String s = new String();
                    System.out.print("\nEnter a string: ");
                    try{
                       s=scanner.nextLine();
                    }
                    catch(Exception e){
                       System.out.println("Invalid input: Quitting...");
                       return;
                    }
                    System.out.println("\nTranslation: "+english2morse(s));
                    break;
            case 3: stop=true;
         }
      }
   }

   // This allows the user to choose an option
   // for the translator
   private static Integer menu()
   {
      Scanner scanner = new Scanner(System.in);

      Integer choice;
      System.out.print("Morse Code Translator"+
                       "\n_____________________"+
                       "\n\t1. Translate character to Morse Code"+
                       "\n\t2. Translate string to Morse Code"+
                       "\n\t3.Exit"+
                       "\nChoice: ");

      // No error checking, just end the program if
      // the user is being annoying
      try{
         choice = scanner.nextInt();
         if(choice < 1 || choice > 3)
            throw new Exception();
      }
      catch(Exception e){
         System.out.println("Invalid input: Quitting...");
         choice=-1;
         System.exit(-1);
      }
      return choice;
   }

   // Translate a string to morse code
   private static String english2morse(String eng)
   {
      String res = new String();
      for(int i=0; i < eng.length();i++)
      {
         switch(Character.toLowerCase(eng.charAt(i)))
         {
            case 'a': res+="·‒";break;
            case 'b': res+="‒···";break;
            case 'c': res+="‒·‒·";break;
            case 'd': res+="‒··";break;
            case 'e': res+="·";break;
            case 'f': res+="··‒·";break;
            case 'g': res+="‒‒·";break;
            case 'h': res+="····";break;
            case 'i': res+="··";break;
            case 'j': res+="·‒‒‒";break;
            case 'k': res+="‒·‒";break;
            case 'l': res+="·‒··";break;
            case 'm': res+="‒‒";break;
            case 'n': res+="‒·";break;
            case 'o': res+="‒‒‒";break;
            case 'p': res+="·‒‒·";break;
            case 'q': res+="‒‒·‒";break;
            case 'r': res+="·‒·";break;
            case 's': res+="···";break;
            case 't': res+="‒";break;
            case 'u': res+="··‒";break;
            case 'v': res+="···‒";break;
            case 'w': res+="·‒‒";break;
            case 'x': res+="‒··‒";break;
            case 'y': res+="‒·‒‒";break;
            case 'z': res+="‒‒··";break;
            default: res+=' ';break;
         }
      }
      return res;
   }

   // Translate a character to morse code
   private static String english2morse(Character eng)
   {
      return english2morse(new String(""+eng));
   }
}
